File: 1st Semester/Computer Fundamentals I/Unit 1/Java/For/Sucesion_Farey.java
/*
 Encontrar la sucesion de farey para un numero, supongamos
 3 = 1/1, 1/2, 1/3
     2/1, 2/2, 2/3
     3/1, 3/2, 3/3
*/

import java.util.Scanner;
public class Sucesion_Farey {
    public static void main (String[] args) {
        int numeroSucesion;
        Scanner input = new Scanner(System.in);
        System.out.println("Introduzca un numero para la sucesion de Farey: ");
        numeroSucesion = input.nextInt();
        for (int i = 1; i <= numeroSucesion; i++) {
            for (int j = 1; j <= numeroSucesion; j++) {
                System.out.printf("%d/%d\t", i, j);
            }
            System.out.println("");
        }
        input.close();
    }
}

File: 1st-semester/computer-fundamentals-1/unit-1/c/for/Sucesion_Farey.c
/*
 Encontrar la sucesion de farey para un numero, supongamos
 3 = 1/1, 1/2, 1/3
     2/1, 2/2, 2/3
     3/1, 3/2, 3/3
*/
#include <stdio.h>
int main () {
    int i, j, numero;
    printf("Introduzca un numero para la secuencia de Farey: "); scanf("%d", &numero); printf("%d = ", numero);
    for (i = 1; i <= numero; i++) {
        for (j = 1; j <= numero; j++) {
            printf("%d/%d", i, j);
            if (i == j && i == numero) printf(". "); // > Se puede comprimir el 'if' porque solo esta imprimiendo, o sea, que solo necesitamos una linea de codigo.
            else printf(", ");
        }
        printf("\n");
    }
    return 0;
}